src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +752,80 @@ const UsersConfigurationRow = ({
     );
 };
 
+// This method needs to be outside of component as re-render would create a new instance of debounce
+const validateKey = debounce(500, (key, setKeyObject) => {
+    cockpit.script(`echo ${key} | ssh-keygen -l -f /dev/stdin`, {})
+            .then(() => {
+                const parts = key.split(" ");
+                if (parts.length > 2) {
+                    setKeyObject({
+                        type: parts[0],
+                        data: parts[1],
+                        comment: parts[2], // comment is optional in SSH-format
+                    });
+                }
+            })
+            .catch(() => setKeyObject(undefined));
+});
+
+const SshKeysRow = ({
+    id, item, onChange, idx, removeitem,
+}) => {
+    const [keyObject, setKeyObject] = useState();
+
+    return (
+        <Grid id={id}>
+            <GridItem span={11}>
+                {keyObject
+                    ? <FlexItem>
+                        <strong>{keyObject.comment}</strong>
+                        <span>{keyObject.comment ? " - " + keyObject.type : keyObject.type}</span>
+                        <div>{keyObject.data}</div>
+                    </FlexItem>
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20231002-121122-826e3047-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)